lib/aggregators/log-aggregator.js

```diff
@@ -27,6 +27,19 @@ class LogAggregator extends EventAggregator {
   }
 
   _toPayloadSync() {
+    if (['starting', 'stopped'].includes(this.agent._state)) {
```

Comment:
from standup: harvest only happen after agent has connected so I don't think we have to check the state.  also with that being said harvest also don't happen until the config is initialized so the very defensive check for the log_event_data key in harvest limits is not necessary you can just access the deep value .